connect_four/board.rb

```diff
@@ -0,0 +1,130 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    @board = Array.new(7) {Array.new(0)} #7 empty columns
+  end
+
+  # Graphic methods
+  def render
+    puts "\n   1 2 3 4 5 6 7"
+    5.downto 0 do |row|
+      print "  |"
+      0.upto 6 do |column|
+        if board[column][row].nil?
+          print " |"
+        else
+          print board[column][row]
+          print "|"
+        end
+      end
+      print "\n"
+    end
+    print "   1 2 3 4 5 6 7\n\n"
+  end
+
+  def insert_coin(column, player_id)
+    if column.nil?
+      false
+    elsif column_is_invalid?(column)
+      puts "Error: That is not a correct column."
+      false
+    elsif column_is_full?(column)
+      puts "Error: The column is full! Please choose another column"
+      false
+    else
+      board[column] << player_id
+      true
+    end
+  end
+
+  def win?
+    # Checks vertical, horizontal, and diaogonal victory conditions
+    vertical_win? ||
+    horizontal_win? ||
+    up_diagonal_win? ||
+    down_diagonal_win?
+  end
+
+  def full?
+    board.each do |column|
+      return false if column.length < 6
+    end
+    return true
+  end
+
+  private
+
+  def column_is_full?(column)
+    board[column].length == 6
+  end
+
+  def column_is_invalid?(column)
```

Comment:
You guys this is so modular. I'm literally loving it. You're gonna rock the testing.